### ISSUE TITLE
Using type as a callable fails

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4258]
**Describe the bug**
```python
from typing import Callable, TypeVar

T = TypeVar("T")

def class_constructor(cls: type[T]) -> Callable[..., T]:
    return cls
```

Recently (1.1.281) this code started having an error of,

```
  /Users/mdrissi/Snapchat/Dev/typing_experiments/type_callable.py:6:12 - error: Expression of type "Type[T@class_constructor]" cannot be assigned to return type "(...) -> T@class_constructor"
    Type "Type[T@class_constructor]" cannot be assigned to type "(...) -> T@class_constructor"
      Type "Type[object]" cannot be assigned to type "(...) -> T@class_constructor"
        Function return type "object" is incompatible with type "T@class_constructor"
```

Before that it worked well.

**VS Code extension or command-line**
pyright cli 1.1.281+ display error. 1.1.280 and earlier showed no errors.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.

**Comment by erictraut** [comment]
Re-opening this bug, as the fix somehow got reverted during a recent sync from pylance.

**Comment by erictraut** [comment]
This is included in pyright 1.1.284, which I just published. It will also be included in a future release of pylance.
